mu4/notation/view/viewmodecontrolmodel.cpp

```diff
@@ -0,0 +1,122 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "viewmodecontrolmodel.h"
+#include "notationtypes.h"
+
+using namespace mu::notation;
+
+ViewModeControlModel::ViewModeControlModel(QObject* parent)
+    : QAbstractListModel(parent)
+{
+    m_roleNames.insert(IdRole, "idRole");
+    m_roleNames.insert(NameRole, "nameRole");
+
+    ViewModeOption page {
+        "Page View", "view-mode-page", ViewMode::PAGE
+    };
+    ViewModeOption continuous {
+        "Continuous View", "view-mode-continuous", ViewMode::LINE
+    };
+    ViewModeOption single {
+        "Single Page", "view-mode-single", ViewMode::SYSTEM
+    };
+    m_viewModeOptions << page << continuous << single;
```

Comment:
I think it'd be more convenient to declare the here like:
`m_viewModeOptions << { "Page View", "view-mode-page", ViewMode::Page" }
                                    << { "Continuous View", "view-mode-continuous", ViewMode::Line }
                                    .......`